src/Nethermind/Nethermind.JsonRpc/Modules/Trace/TraceRpcModule.cs

```diff
@@ -144,15 +152,28 @@ private SearchResult<BlockHeader> SearchBlockHeaderForTraceCall(BlockParameter b
             return new SearchResult<BlockHeader>(header);
         }
 
-        private ResultWrapper<ParityTxTraceFromReplay> TraceTx(Transaction tx, string[] traceTypes, BlockParameter blockParameter)
+        private ResultWrapper<ParityTxTraceFromReplay> TraceTx(Transaction tx, string[] traceTypes, BlockParameter blockParameter,
+            Dictionary<Address, AccountOverride>? stateOverride = null)
         {
             SearchResult<BlockHeader> headerSearch = SearchBlockHeaderForTraceCall(blockParameter);
             if (headerSearch.IsError)
             {
                 return ResultWrapper<ParityTxTraceFromReplay>.Fail(headerSearch);
             }
 
-            Block block = new(headerSearch.Object!, new[] { tx }, Enumerable.Empty<BlockHeader>());
+            BlockHeader header = headerSearch.Object!.Clone();
+
+            // Fake next-block header to force BlockProcessor to start from <blockParameter> instead of <blockParameter> - 1
+            // TODO consider alternative approaches to achieve this
+            BlockHeader nextHeader = header.Clone();
```

Comment:
`TraceBlock` implementation called below will rollback to the previous block before starting processing, as needed for most of the `trace_*` methods. This code creates a fake "next" header, to actually enforce executing `tx` to use thestate from the same block as provided in `blockParameter` for `trace_call` and `trace_rawTransaction`.

This doesn't look ideal and may break with changes to `BlockProcessor`, but it gets the job done, without a lot of refactoring.

One of the alternative approaches can be to extend `processingOptions`, yet it may be a bit cumbersome.